src/main/java/de/hysky/skyblocker/skyblock/tabhud/config/WidgetsListTab.java

```diff
@@ -0,0 +1,209 @@
+
+
+	public void setCustomWidgetEntries(Collection<WidgetEntry> entries) {
+		this.customWidgetEntries.clear();
+		this.customWidgetEntries.addAll(entries);
+		listNeedsUpdate = true;
+	}
+
+	public List<WidgetEntry> getCustomWidgetEntries() {
+		return customWidgetEntries;
+	}
+
+	public boolean listNeedsUpdate() {
+		boolean b = listNeedsUpdate;
+		listNeedsUpdate = false;
+		return b;
+	}
+
+	public ObjectSet<Int2ObjectMap.Entry<WidgetsListSlotEntry>> getEntries() {
+		return entries.int2ObjectEntrySet();
+	}
+
+	public WidgetsListTab(MinecraftClient client, @Nullable GenericContainerScreenHandler handler) {
+        widgetsElementList = new WidgetsElementList(this, client, 0, 0, 0);
+        this.client = client;
+        this.handler = handler;
+        back = ButtonWidget.builder(Text.literal("Back"), button -> clickAndWaitForServer(48, 0))
+                .size(64, 15)
+                .build();
+        thirdColumnButton = ButtonWidget.builder(Text.literal("3rd Column:"), button -> clickAndWaitForServer(50, 0))
+                .size(120, 15)
+                .build();
+        thirdColumnButton.setTooltip(Tooltip.of(Text.literal("It is recommended to have this enabled, to have more info be displayed!")));
+        previousPage = ButtonWidget.builder(Text.literal("Previous Page"), button -> clickAndWaitForServer(45, 0))
+                .size(100, 15)
+                .build();
+        nextPage = ButtonWidget.builder(Text.literal("Next Page"), button -> clickAndWaitForServer(53, 0))
+                .size(100, 15)
+                .build();
+		if (handler == null) {
+			back.visible = false;
+			previousPage.visible = false;
+			nextPage.visible = false;
+			thirdColumnButton.visible = false;
+		}
+    }
+
+    @Override
+    public Text getTitle() {
+        return Text.literal("Widgets");
+    }
+
+    @Override
+    public void forEachChild(Consumer<ClickableWidget> consumer) {
+        consumer.accept(back);
+        consumer.accept(previousPage);
+        consumer.accept(nextPage);
+        consumer.accept(thirdColumnButton);
+        consumer.accept(widgetsElementList);
+    }
+
+    public void clickAndWaitForServer(int slot, int button) {
+        if (waitingForServer || handler == null) return;
+        if (client.interactionManager == null || this.client.player == null) return;
+        client.interactionManager.clickSlot(handler.syncId, slot, button, SlotActionType.PICKUP, this.client.player);
+        waitingForServer = true;
+    }
+
+    public void shiftClickAndWaitForServer(int slot, int button) {
+        if (waitingForServer || handler == null) return;
+        if (client.interactionManager == null || this.client.player == null) return;
+        client.interactionManager.clickSlot(handler.syncId, slot, button, SlotActionType.QUICK_MOVE, this.client.player);
+        // When moving a widget down it gets stuck sometimes
+        Scheduler.INSTANCE.schedule(() -> this.waitingForServer = false, 1);
```

Comment:
Is one tick going to be enough here? Maybe 20 ticks to be safe? Why use shift-click anyways, just use normal click?

Reply:
didn't have any issues with one tick. Shift click is needed to change the ordering of the tab widgets in the hypixel GUI, that's just how it works 🤷

Reply:
I'm just a little worried about people with like 200ms or even higher ping.

Reply:
I have constant 120 ping and i got no issues, but i could increase it to 4 ticks